pandas/io/parsers.py

```diff
@@ -1605,9 +1607,18 @@ def _cast_types(self, values, cast_type, column):
             # XXX this is for consistency with
             # c-parser which parses all categories
             # as strings
-            if not is_object_dtype(values):
+            known_cats = (isinstance(cast_type, CategoricalDtype) and
+                          cast_type.categories is not None)
+
+            categories = ordered = None
```

Comment:
why is this not using ``Categorical._inferred_from_categories``, this code duplication is just making technical debt.

Reply:
I'm not sure how much cleaner 3de75cd is. This really don't share much code, since the python parser has values, while the C parser has categories and codes. And the python parser has to maybe cast values to strings with `cast_type='category'`.